reviews/review-01+2020-11-02.md

```diff
@@ -0,0 +1,305 @@
+    checking.
+
+    -   Good examples:
+
+        -   Introduction to chapter 3 (Going Further with the Unix
+            Shell): "This chapter extends these ideas to show how we can
+            create new tools by saving commands in files and how to use
+            a more powerful version of wildcards to extract data from
+            files."
+
+            -   Topic: We're going to put commands in files and use
+                wildcards
+
+            -   Why: We can create new tools and get data (and do better
+                work)
+
+        -   3.2, using variables in a script: "Getting the name of the
+            author for only one of the books isn't all that useful. What
+            we really want is a way to get the name of the author from
+            any of our files." (and then go on to talk about variables)
+
+            -   Topic: Using variables
+
+            -   Why: We want to solve the problem of analyzing all of
+                our data.
+
+    -   Examples that could be improved:
+
+        -   2.1, starting to use the shell: I'm looking for something
+            like "Our first shell commands will allow us to navigate and
+            view the contents of our file system, a foundational skill
+            for using the shell to run other commands and use files
+            organized in a project."
+
+        -   3.6, introducing shell variables: I'm missing the final "so
+            what"---like "knowing about shell variables allows us to
+            customize our shell working environment if we use it
+            frequently."
+
+    -   If you only review one chapter for these "topic sentences" I
+        would recommend chapter 7. The content in that chapter is
+        crucially important, and currently can read as somewhat
+        unconnected topics around people and project management. A topic
+        sentence in each section that links back to the goal of
+        empowering people to collaborate together and in a safe and
+        productive manner (or connecting a particular section to a
+        previous section in the chapter, e.g. 7.6 is really a follow on
+        from 7.5) would go a long way towards helping that material
+        cohere. That, or adding more details to the introduction,
+        outlining the major topics that will be covered in the chapter.
+
+-   I had the most difficulty in following chapter 10 (Error Handling);
+    it was the one chapter where I felt like I didn't have a sufficient
+    mental model to understand the details of the chapter. I'm not sure
+    whether the solution is for me to have a better understanding of
+    Python or for the chapter to more explicitly generate a mental model
+    for errors and exceptions.
+
+-   For some of the initial appendices (A, B, C), it wasn't clear if the
+    text was meant to be used as a sample for your own project (or the
+    zipfs sample project) or as a reference for the project that is the
+    manuscript itself.
```

Comment:
With respect to the three Appendices in question:
- **Appendix A: License**: This is the license for our book and is referred to in the introduction chapter, so it needs to stay (maybe we could make it clearer in the text at the beginning of that appendix that it is not an example license but is the actual license for the book)
- **Appendix B: Code of Conduct**: This is the code of conduct for contributing to the actual book. It is not referred anywhere in the book. We provide an example Code of Conduct in full in Section 7.3, so we don't need one in an appendix as an example. I'd suggest we can remove Appendix B from the book as it only needs to exist in our GitHub repo (because that's where people will be making contributions).
- Appendix C: Contributing: This is the contributing guide for our book and is not referred to anywhere in the book. I think it only needs to exist in the GitHub repo. In fact, in Exercise 7.14.1 we ask the reader to go to our GitHub repo to find the contributing information.

Reply:
For item one on the licensing, an alternative approach would be to split Chapter 1 into "Overview" or "Welcome" where we talk about the big picture, how to use the material, licensing, acknowledgements, how to contribute to the book (link to GitHub contributing) and code of conduct for contributing (link to GitHub), etc. Then we can delete Appendix A and C, and revamp Appendix B into Code of Conduct for participating in the course (or at least have it as a template for instructors to use). And have a new Chapter called "Syllabus" where it has Audience, Project structure, and syllabus.

Reply:
@lwjohnst86 So "Welcome" would essentially be the numberless prologue mentioned earlier and "Syllabus" would chapter 1?

Reply:
@DamienIrving yes, exactly!

Reply:
I will pull the 'Contributing' out - PR coming.